examples_gbs/run_tutorial_similarity.py

```diff
@@ -0,0 +1,278 @@
+# 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0]`` and has orbit:
+
+print(similarity.sample_to_orbit(m0[0]))
+
+##############################################################################
+# Other samples can be randomly generated from the ``[1, 1]`` orbit using:
+
+print(similarity.orbit_to_sample([1, 1], modes=m0.modes))
+
+##############################################################################
+# Orbits provide a useful way to coarse grain the samples from GBS into outcomes that are
+# statistically more likely to be observed. However, we are interested in coarse graining further
+# into **events**, which correspond to a combination of orbits with the same photon number such
+# that the number of photons counted in each mode does not exceed a fixed value
+# ``max_count_per_mode``. To understand this, let's look at all of the orbits with a photon
+# number of 5:
+
+print(list(similarity.orbits(5)))
+
+##############################################################################
+# All 5-photon samples fall into one of the orbits above. A 5-photon event with
+# ``max_count_per_mode = 3`` means that we include the orbits: ``[[1, 1, 1, 1, 1], [2, 1, 1, 1],
+# [3, 1, 1], [2, 2, 1], [3, 2]]`` and ignore the orbits ``[[4, 1], [5]]``. For example,
+# the sample ``[0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 3, 0, 0, 0]`` is a 5-photon event:
+
+print(similarity.sample_to_event([0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 3, 0, 0, 0], 3))
+
+##############################################################################
+# Samples with more than ``max_count_per_mode`` in any mode are not counted as part of the event:
+
+print(similarity.sample_to_event([0, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0], 3))
+
+##############################################################################
+# Now that we have mastered orbits and events, how can we make a feature vector? It was shown in
+# :cite:`schuld2019quantum` that one way of making a feature vector of a graph is to measure the
+# probabilities of events. Specifically, for a :math:`k` photon event :math:`E_{k, n_{\max}}`
+# with maximum count per mode :math:`n_{\max}` and corresponding probability :math:`p_{k,
+# n_{\max}}:=p_{E_{k, n_{\max}}}(G)` with respect to a graph :math:`G`, a feature vector can be
+# written as
+#
+# .. math::
+#     f_{\mathbf{k}, n_{\max}} = (p_{k_{1}, n_{\max}}, p_{k_{2}, n_{\max}}, \ldots , p_{k_{K},
+#         n_{\max}}),
+#
+# where :math:`\mathbf{k} := (k_{1}, k_{2}, \ldots , k_{K})` is a list of different total photon
+# numbers.
+#
+# Calculating a feature vector
+# ----------------------------
+#
+# We provide two methods for calculating a feature vector of GBS event probabilities in
+# Strawberry Fields:
+#
+# 1. Through sampling
+# 2. Using a Monte Carlo estimate of the probability
+#
+# For the first method, all one needs to do is generate some GBS samples from the graph of

```

Comment:
Probably good to say say first that we'll focus on three-dimensional feature vectors, which can be calculated as follows

Reply:
Done